worldedit-core/src/main/java/com/sk89q/worldedit/LocalSession.java

```diff
@@ -573,17 +573,15 @@ public EditSession undo(@Nullable BlockBag newBlockBag, Actor actor) {
         loadSessionHistoryFromDisk(actor.getUniqueId(), world);
         if (getHistoryNegativeIndex() < history.size()) {
             ChangeSet changeSet = getChangeSet(history.get(getHistoryIndex()));
-            EditSessionBuilder builder = new EditSessionBuilder(world)
+            try (EditSession newEditSession = WorldEdit.getInstance().newEditSessionBuilder().world(world)
+                .allowedRegionsEverywhere()
```

Comment:
This should not be a thing. If there is a diff with WE, then there will have to be a diff with WE, otherwise we can be changing blocks in regions the player no longer has access to.